rasterio/rio/warp.py

```diff
@@ -135,8 +135,7 @@ def warp(
                         raise click.BadParameter('Required when using --bounds',
                             param='res', param_hint='res')
 
-                    xmin, ymin, xmax, ymax = transform_bounds(src.crs, dst_crs,
-                                                              *bounds)
+                    xmin, ymin, xmax, ymax = bounds
```

Comment:
This doesn't fix #541 but it does fix yet unreported bugs: `bounds` is in the dst crs already, not src crs.